### PR TITLE
actionified, formatted and doxy'd

### PR DESCRIPTION
**Pull request by siddacious**

### CODE REVIEW

**Comment by ladyada** [comment]
please add skips :)

File: Adafruit_FONA.cpp [hunk 1]
```diff
@@ -14,23 +14,26 @@
...
+// software serial and are currently incompatible with the FONA library).
+#include <algorithm>
+using namespace std;
 #endif
-
-
-Adafruit_FONA::Adafruit_FONA(int8_t rst)
-{
+/**
+ * @brief Construct a new Adafruit_FONA::Adafruit_FONA object
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: Construct a new Adafruit_FONA object

File: Adafruit_FONA.cpp [hunk 2]
```diff
@@ -42,11 +45,19 @@ Adafruit_FONA::Adafruit_FONA(int8_t rst)
...
+/**
+ * @brief Get the module type
+ *
+ * @return uint8_t The module type
+ */
+uint8_t Adafruit_FONA::type(void) { return _type; }
+/**
+ * @brief Connect to the cell module
+ *
+ * @param port the serial connection to uset o connect
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: use to

File: Adafruit_FONA.cpp [hunk 3]
```diff
@@ -97,103 +110,166 @@ boolean Adafruit_FONA::begin(Stream &port) {
...
 
-
 /********* Serial port ********************************************/
-boolean Adafruit_FONA::setBaudrate(uint16_t baud) {
+
+/**
+ * @brief  Set the baud rate that the module will use
+ *
+ * @param baud The new baud rate to set
+ * @return true success
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: only one @return

File: Adafruit_FONA.cpp [hunk 4]
```diff
@@ -97,103 +110,166 @@ boolean Adafruit_FONA::begin(Stream &port) {
...
   return sendCheckReply(F("AT+IPREX="), baud, ok_reply);
 }
 
 /********* Real Time Clock ********************************************/
 
-boolean Adafruit_FONA::readRTC(uint8_t *year, uint8_t *month, uint8_t *date, uint8_t *hr, uint8_t *min, uint8_t *sec) {
+/**
+ * @brief Read the Real Time Clock
+ *
+ * @param year
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: specify they are pointers

File: Adafruit_FONA.cpp [hunk 5]
```diff
@@ -97,103 +110,166 @@ boolean Adafruit_FONA::begin(Stream &port) {
...
 
   DEBUG_PRINTLN(*year);
 }
-
-boolean Adafruit_FONA::enableRTC(uint8_t i) {
-  if (! sendCheckReply(F("AT+CLTS="), i, ok_reply))
+/**
+ * @brief Enable the Real Time Clock
+ *
+ * @param i
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: explain parameters

File: Adafruit_FONA.cpp [hunk 6]
```diff
@@ -97,103 +110,166 @@ boolean Adafruit_FONA::begin(Stream &port) {
...
+ * @return true success
+ * @return false failure
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: only one return line

File: Adafruit_FONA.cpp [hunk 7]
```diff
@@ -97,103 +110,166 @@ boolean Adafruit_FONA::begin(Stream &port) {
...
     return false;
   return sendCheckReply(F("AT&W"), ok_reply);
 }
 
-
 /********* BATTERY & ADC ********************************************/
 
-/* returns value in mV (uint16_t) */
-boolean Adafruit_FONA::getBattVoltage(uint16_t *v) {
+/* returns  */
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: remove

File: Adafruit_FONA.cpp [hunk 8]
```diff
@@ -97,103 +110,161 @@ boolean Adafruit_FONA::begin(Stream &port) {
...
+ * @param hr pointer to a uint8_t to be set with hour data
+ * @param min pointer to a uint8_t to be set with minute data
+ * @param sec pointer to a uint8_t to be set with year data
+ * @return true: success, false: failure
+ */
+bool Adafruit_FONA::readRTC(uint8_t *year, uint8_t *month, uint8_t *day,
+                            uint8_t *hr, uint8_t *min, uint8_t *sec) {
   uint16_t v;
   sendParseReply(F("AT+CCLK?"), F("+CCLK: "), &v, '/', 0);
+  // TODO: https://github.com/adafruit/Adafruit_FONA/issues/111
```
Comment (siddacious) [inline Adafruit_FONA.cpp]: @ladyada Pretty sure you're aware but this doesn't work fully

Comment (ladyada) [reply Adafruit_FONA.cpp]: then add docyumentation that it is not compelte

**Comment by siddacious** [comment]
@ladyada done and passing

File: Adafruit_FONA.cpp [hunk 9]
```diff
@@ -204,12 +277,18 @@ uint8_t Adafruit_FONA::unlockSIM(char *pin)
   return sendCheckReply(sendbuff, ok_reply);
 }
 
+/**
+ * @brief Get the SIM CCID
+ *
+ * @param ccid pointer to ccid buffer
```
Comment (ladyada) [inline Adafruit_FONA.cpp]: please Go thru the doxygen again to make sure every function has good documentation.

**Comment by siddacious** [comment]
@ladyada this is ready for your review. I did my best to figure out what each fn was doing based on reading the code and the docs for the AT command set but I may have misunderstood something.

**Pull request merged**